app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx

```diff
@@ -245,49 +270,45 @@ export default function CompetingStep({
       )}
 
       <>
-        {!competitionInfo['registration_opened?'] && (
-          <Message warning>
-            {i18n.t('competitions.registration_v2.register.early_registration')}
-          </Message>
-        )}
-
         {hasPaid && (
           <Message success>
             {i18n.t('competitions.registration_v2.register.already_paid', { comp_name: competitionInfo.name })}
           </Message>
         )}
 
-        <Form onSubmit={handleSubmit}>
-          <Form.Field required>
+        <Form onSubmit={handleSubmit} warning={formWarnings.length > 0} size="large">
+          <Message
+            warning
+            list={formWarnings}
+          />
+          <Form.Field required error={selectedEvents.length === 0}>
             <EventSelector
               onEventSelection={handleEventSelection}
               eventList={competitionInfo.event_ids}
               selectedEvents={selectedEvents}
               id="event-selection"
               maxEvents={maxEvents}
             />
-            <p
-              dangerouslySetInnerHTML={{
-                __html: i18n.t(!competitionInfo.events_per_registration_limit ? 'registrations.preferred_events_prompt_html' : 'competitions.registration_v2.register.event_limit', {
-                  link: `<a href="${userPreferencesRoute}">here</a>`,
-                  max_events: competitionInfo.events_per_registration_limit,
-                }),
-              }}
-            />
+            {!competitionInfo.events_per_registration_limit
+              && (
+              <p
+                dangerouslySetInnerHTML={{
+                  __html: i18n.t('registrations.preferred_events_prompt_html', {
+                    link: `<a href="${userPreferencesRoute}">here</a>`,
+                  }),
+                }}
+              />
```

Comment:
We have an `I18nHtmlTranslate` component to avoid repeating this `dangerouslySetInnerHTML` idiom (with some additional sanitization)

Reply:
cool!